irr/src/CMeshManipulator.h

```diff
@@ -29,6 +29,12 @@ class CMeshManipulator : public IMeshManipulator
 	\param smooth: Whether to use smoothed normals. */
 	void recalculateNormals(IMeshBuffer *buffer, bool smooth = false, bool angleWeighted = false) const override;
 
+	//! Recalculates tangents, requires a tangent mesh buffer
+	virtual void recalculateTangents(IMeshBuffer* buffer, bool recalculateNormals = false, bool smooth = false, bool angleWeighted = false) const _IRR_OVERRIDE_;
```

Comment:
What are the tangents/normals used for? (Sorry wasn't clear from looking at the rest of the code)

Reply:
They‘re required for bump mapping